### PR TITLE
Updated test suite and added test docs

### PR DESCRIPTION
**Pull request by andythorne**
### Short Description:

Updated test suite to use mock classes as `__getattr__` could not be overridden
### Changes:
- Refactored test suite
- Added testing docs

@OpenPoGo/maintainers

### CODE REVIEW

File: docs/testing.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Testing
+
+Testing in OpenPoGoBot is preformed by the [`pytest`](http://doc.pytest.org/en/latest/),
```
Comment (MaxLeiter) [inline docs/testing.md]: Performed*

Comment (andythorne) [inline docs/testing.md]: Fixed :)

**Comment by tehp** [comment]
👍 
Thanks!

[Approved with PullApprove](https://pullapprove.com/OpenPoGo/OpenPoGoBot/pull-request/248/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=OpenPoGo/OpenPoGoBot)

**Comment by MaxLeiter** [comment]
Waiting on Travis for merge

:+1:

[Approved with PullApprove](https://pullapprove.com/OpenPoGo/OpenPoGoBot/pull-request/248/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=OpenPoGo/OpenPoGoBot)

**Comment by MaxLeiter** [comment]
leaving for you to merge, @andythorne

**Pull request merged**